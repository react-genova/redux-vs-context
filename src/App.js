import React from "react";
import styled from "styled-components";
// Common
import Title from "./components/Title";
import Subtitle from "./components/Subtitle";
// Context
import AppContext from "./_context/AppContext";
import PlayerContainerContext from "./player/context/PlayerContainer";
import NameContainerContext from "./player/context/NameContainer";
import SurnameContainerContext from "./player/context/SurnameContainer";
// Redux
import AppRedux from "./_redux/AppRedux";
import PlayerContainerRedux from "./player/redux/PlayerContainer";
import NameContainerRedux from "./player/redux/NameContainer";
import SurnameContainerRedux from "./player/redux/SurnameContainer";

const App = ({ className }) => (
    <div className={className}>
        <div className="AppContext">
            <Title title="Context" />
            <AppContext>
                <Subtitle subtitle="Player" />
                <PlayerContainerContext />
                <NameContainerContext />
                <SurnameContainerContext />
            </AppContext>
        </div>
        <div className="AppRedux">
            <Title title="Redux" />
            <AppRedux>
                <Subtitle subtitle="Player" />
                <PlayerContainerRedux />
                <NameContainerRedux />
                <SurnameContainerRedux />
            </AppRedux>
        </div>
    </div>
);

const AppStyled = styled(App)`
    display: flex;
    .AppContext {
        padding: 0 10px;
        display: flex;
        flex-direction: column;
        flex: 1;
        background-color: cadetblue;
    }
    .AppRedux {
        padding: 0 10px;
        display: flex;
        flex-direction: column;
        flex: 1;
        background-color: darkcyan;
    }
`;

export default AppStyled;
