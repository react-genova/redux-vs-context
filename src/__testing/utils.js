export const mockDefault = mock => ({
    __esModule: true,
    default: mock,
});
