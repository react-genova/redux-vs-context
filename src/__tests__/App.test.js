import React from 'react';
import { render } from 'react-testing-library';
import { mockDefault } from '../__testing/utils';
import App from '../App';
import 'jest-styled-components';

jest.mock('react-redux');
jest.mock('../player/context/PlayerContainer', () => mockDefault(() => 'ContextPlayerContainer'));
jest.mock('../player/context/NameContainer', () => mockDefault(() => 'ContextNameContainer'));
jest.mock('../player/context/SurnameContainer', () => mockDefault(() => 'ContextSurnameContainer'));
jest.mock('../player/redux/PlayerContainer', () => mockDefault(() => 'ReduxPlayerContainer'));
jest.mock('../player/redux/NameContainer', () => mockDefault(() => 'ReduxNameContainer'));
jest.mock('../player/redux/SurnameContainer', () => mockDefault(() => 'ReduxSurnameContainer'));

describe('App main component', () => {
    global.console = { log: () => null };
    it('renders without crashing', () => {
        const { container } = render(<App />);
        expect(container.firstChild).toMatchSnapshot();
    });
});