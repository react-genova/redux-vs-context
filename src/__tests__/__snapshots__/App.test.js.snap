// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App main component renders without crashing 1`] = `
.c1 {
  text-align: center;
  margin: 10px 0 0 0;
  color: #afa;
}

.c2 {
  margin: 0 10px;
  text-align: left;
  text-transform: capitalize;
  color: #8d8;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 .AppContext {
  padding: 0 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  background-color: cadetblue;
}

.c0 .AppRedux {
  padding: 0 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  background-color: darkcyan;
}

<div
  class="c0"
>
  <div
    class="AppContext"
  >
    <h1
      class="c1"
    >
      Context
    </h1>
    <h2
      class="c2"
    >
      Player
    </h2>
    ContextPlayerContainer
    ContextNameContainer
    ContextSurnameContainer
  </div>
  <div
    class="AppRedux"
  >
    <h1
      class="c1"
    >
      Redux
    </h1>
    <h2
      class="c2"
    >
      Player
    </h2>
    ReduxPlayerContainer
    ReduxNameContainer
    ReduxSurnameContainer
  </div>
</div>
`;
